baton-maven-plugin/src/main/java/org/technologybrewery/baton/BatonMojo.java

```diff
@@ -0,0 +1,236 @@
+            }
+
+            // add common root files:
+            FileSet rootFileSets = new FileSet();
+            rootFileSets.setDirectory(baseDirectory.getAbsolutePath());
+            rootFileSets.addInclude("pom.xml");
+            rootFileSets.addInclude("*.toml");
+            defaultFileSets.add(rootFileSets);
+            getLog().debug(String.format("\tAdded: %s with includes: %s", rootFileSets.getDirectory(), rootFileSets.getIncludes()));
+
+            fileSets = defaultFileSets.toArray(new FileSet[0]);
+
+            getLog().debug("Default file set definition completed");
+        }
+    }
+
+    @Override
+    public void execute() throws MojoExecutionException, MojoFailureException {
+        loadMigrations();
+
+        defaultFileSets();
+
+        BatonExecutionSummary summary = performMigration(migrations.values());
+        getLog().info(summary.getSummary());
+
+    }
+
+    BatonExecutionSummary performMigration(Collection<MigrationTarget> targets) {
+        BatonExecutionSummary executionSummary = new BatonExecutionSummary();
+        for (MigrationTarget target : targets) {
+            try {
+                getLog().debug(String.format("Executing Migration: %s (%s)", target.getName(), target.getImplementation()));
+                Class<Migration> implementationClass = (Class<Migration>) Class.forName(target.getImplementation());
+                Constructor<Migration> constructor = implementationClass.getConstructor();
+                Migration migration = constructor.newInstance();
+                FileSet[] migrationSpecificFileSets = (CollectionUtils.isNotEmpty(target.getFileSets())) ? getFileSetsForTarget(target) : fileSets;
+                MigrationSummary migrationSummary = migration.execute(migrationSpecificFileSets);
+                executionSummary.addMigrationSummary(migrationSummary);
+
+            } catch (Exception e) {
+                throw new BatonException("Could not complete migrations!", e);
+            }
+        }
+
+        return executionSummary;
+    }
+
+    /**
+     * Scans the classpath for any migrations.json files and loads all defined {@link MigrationTarget} configurations.
+     */
+    protected void loadMigrations() {
+        Enumeration<URL> migrationsEnumeration = null;
+        try {
+            migrationsEnumeration = getClass().getClassLoader().getResources(migrationsFileName);
+
+        } catch (IOException ioe) {
+            throw new BatonException("Unable to find migrations!", ioe);
+        }
+
+        URL migrationsResource;
+        while (migrationsEnumeration.hasMoreElements()) {
+            migrationsResource = migrationsEnumeration.nextElement();
+            getLog().info(String.format("Loading migrations from: %s", migrationsResource.toString()));
+
+            try (InputStream migrationsStream = migrationsResource.openStream()) {
+                File tempMigrationsFile = File.createTempFile("migrations", ".json");
+                Files.copy(
```

Comment:
Q: Why not just read from the stream directly?

Reply:
The json schema library requires us to read the file twice - so with a stream, there is no way to re-read the file for validation.